src/Math-TSNE/PMTSNE.class.st

```diff
@@ -198,23 +203,23 @@ PMTSNE >> perplexityDefaultValue [
 
 { #category : #running }
 PMTSNE >> runPcaOnX [
-	"Runs PCA on X in order to reduce its dimensionality to initialdims dimensions.
-
-	print ""Preprocessing the data using PCA...""
-	(n, d) = X.shape;
-	X = X - Math.tile(Math.mean(X, 0), (n, 1));
-	(l, M) = Math.linalg.eig(Math.dot(X.T, X));
-	Y = Math.dot(X, M[:,0:no_dims]);
-	return Y;
-"
+	"Runs PCA on X in order to reduce its dimensionality to initialDims dimensions."
 
-	"| analyzer i |
-	analyzer := PMPrincipalComponentAnalyser new: initialdims.
-	i := 1.
-	x dimension x
-		timesRepeat: [ analyzer accumulate: (x rowAt: i).
-			i := i + 1 ].
-	^ analyzer components"
+	self runPcaOnXUsing: PMPrincipalComponentAnalyserJacobiTransformation.
+]
+
+{ #category : #running }
+PMTSNE >> runPcaOnXUsing: aClass [
+	"Runs aClass PCA on X in order to reduce its dimensionality to initialDims dimensions."
+
+	| scaler pca |
+	aClass superclass = PMPrincipalComponentAnalyser
```

Comment:
Why do we check that aClass is a subclass of PMPrincipalComponentAnalyser ?
I would remove this test.

Reply:
Yes, I'll remove it.

Reply:
Resolved in https://github.com/PolyMathOrg/PolyMath/pull/117/commits/7c69a7fc998e4271a4bbf8b26ec3ca65b9a3a91c
(This commit will be squashed)